### PR TITLE
some zipkin fix

### PR DESCRIPTION
**Pull request by songrgg**
1. remove a blocking risk when kafka is offline and the span channel is full
2. remove parentID when parentID is 0 for [openzipkin/zipkin](https://github.com/openzipkin/zipkin) doesn't adjust the time skew between services when parentID is not null. see: [CorrectForClockSkew.java](https://github.com/openzipkin/zipkin/blob/483f78ed3546b3d28e2555c50381eebf67bbf82d/zipkin/src/main/java/zipkin/internal/CorrectForClockSkew.java#L45)
3. fix the wrong mapping. referred to http://zipkin.io/pages/instrumenting.html, zipkin chains should be client start -> server receive -> server send -> client receive

zipkin works well after I applied these fixes.

### CODE REVIEW

File: trace/zipkin/zipkin.go [hunk 1]
```diff
@@ -91,14 +91,26 @@ func toEndpoint(s *registry.Service) *zipkincore.Endpoint {
...
-	span := &zipkincore.Span{
-		TraceID:   toInt64(s.TraceId),
-		Name:      s.Name,
-		ID:        toInt64(s.Id),
-		ParentID:  thrift.Int64Ptr(toInt64(s.ParentId)),
-		Debug:     s.Debug,
-		Timestamp: thrift.Int64Ptr(s.Timestamp.UnixNano() / 1e3),
-		Duration:  thrift.Int64Ptr(s.Duration.Nanoseconds() / 1e3),
+	var span *zipkincore.Span
+	if parentID := toInt64(s.ParentId); parentID != 0 {
```
Comment (asim) [inline trace/zipkin/zipkin.go]: This is a seriously ugly if else condition. Please don't do this. Create the span as usual, then in less than 3 lines set the ParentID if its not 0.

File: trace/zipkin/zipkin.go [hunk 2]
```diff
@@ -214,7 +226,11 @@ func (z *zipkin) Close() error {
 }
 
 func (z *zipkin) Collect(s *trace.Span) error {
-	z.spans <- s
+	select {
+	case z.spans <- s:
+	default:
```
Comment (asim) [inline trace/zipkin/zipkin.go]: Return an error rather than logging.

**Comment by songrgg** [comment]
updated

File: trace/zipkin/zipkin.go [hunk 3]
```diff
@@ -229,7 +221,7 @@ func (z *zipkin) Collect(s *trace.Span) error {
 	select {
 	case z.spans <- s:
 	default:
-		log.Println("z.spans channel is full, discading data")
+		return errors.New("zinpin span channel is full")
```
Comment (asim) [inline trace/zipkin/zipkin.go]: spelling mistake

**Pull request merged**